tests/org.eclipse.ui.workbench.texteditor.tests/src/org/eclipse/ui/findandreplace/FindReplaceLogicTest.java

```diff
@@ -0,0 +1,296 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Vector Informatik GmbH and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Vector Informatik GmbH - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.ui.findandreplace;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.swt.widgets.Shell;
+
+import org.eclipse.jface.text.Document;
+import org.eclipse.jface.text.IFindReplaceTarget;
+import org.eclipse.jface.text.TextViewer;
+
+public class FindReplaceLogicTest {
```

Comment:
When writing new tests, it would be good to use hamcrest matchers instead of the basic JUnit assert*() functionality. This is particularly useful for replacing some assertTrue() statements, which will only fail with "expected true, but was false". Consider for example `assertTrue(a instance b)  compared to `assertThat(a, instanceOf(b))`, which will tell you exactly which type was expected and which type the object had. Also using `assertThat(a, is(b))` is (in my opinion) more comprehensible (and more powerful) than `assertEquals(b, a)`, at least because I am always confused wheter the first argument is the actual or the expected one (and that even changed between JUnit 4 and 5).